sqlmesh/core/audit/definition.py

```diff
@@ -247,22 +261,173 @@ def macro_definitions(self) -> t.List[d.MacroDef]:
+            }
+        elif node.is_audit:
+            node = t.cast(StandaloneAudit, node)
+            node_dialect = node.dialect
+            kwargs = {"python_env": node.python_env}
+
         return QueryRenderer(
             self.query,
-            self.dialect or model.dialect,
+            self.dialect or node_dialect,
             self.macro_definitions,
             path=self._path or Path(),
             jinja_macro_registry=self.jinja_macros,
-            python_env=model.python_env,
-            only_execution_time=model.kind.only_execution_time,
+            **kwargs,
         )
 
 
+class StandaloneAudit(_Node):
+    """
+    Args:
+        name: The unique name of the standalone audit.
+        owner: The owner of the standalone audit.
+        description: The optional standalone audit description.
+        stamp: An optional arbitrary string sequence used to create new standalone audit versions without making
+            changes to any of the functional components of the definition.
+        tags:
+        hash_raw_query:
+    """
+
+    name: str
+    depends_on_: t.Optional[t.Set[str]] = Field(default=None, alias="depends_on")
+    audit: Audit
+    tags: t.List[str] = []
+    hash_raw_query: bool = False
+
+    python_env_: t.Optional[t.Dict[str, Executable]] = Field(default=None, alias="python_env")
+    source_type: Literal["audit"] = "audit"
+
+    _path: Path = Path()
+    _depends_on: t.Optional[t.Set[str]] = None
+
+    @model_validator(mode="before")
+    @classmethod
+    def validate_fields(
+        cls, values: t.Dict[str, t.Union[t.Tuple[str, ...], t.Optional[str], t.Dict[str, t.Any]]]
+    ) -> t.Dict[str, t.Union[t.Tuple[str, ...], t.Optional[str], t.Dict[str, t.Any]]]:
+        if not "name" in values and "audit" in values:
+            values["name"] = t.cast(Audit, values["audit"]).name
+        return values
+
+    @property
+    def depends_on(self) -> t.Set[str]:
+        if self._depends_on is None:
+            self._depends_on = self.depends_on_ or set()
+
+            query = self.audit.render_query(self)
+            if query is not None:
+                self._depends_on |= d.find_tables(query, dialect=self.dialect)
+
+            self._depends_on -= {self.name}
+        return self._depends_on
+
+    @property
+    def python_env(self) -> t.Dict[str, Executable]:
+        return self.python_env_ or {}
+
+    @property
+    def sorted_python_env(self) -> t.List[t.Tuple[str, Executable]]:
+        """Returns the python env sorted by executable kind and then var name."""
+        return sorted(self.python_env.items(), key=lambda x: (x[1].kind, x[0]))
+
+    @property
+    def dialect(self) -> str:
+        return self.audit.dialect
+
+    @property
+    def data_hash(self) -> str:
+        """
+        Computes the data hash for the node.
+
+        Returns:
+            The data hash for the node.
+        """
+        # StandaloneAudits do not have a data hash
+        return ""
```

Comment:
This should be either `"0"` or `hash_data("")`.

Reply:
The method returns a string. I used `""` to indicate there is no data hash vs a static data dash that `"0" `or `hash_data("") `would produce. Are you thinking there is no value in indicating there isn't a data hash?